lib/cargo.js

```diff
@@ -105,6 +106,18 @@ export function provideBuilder() {
         }
       }
 
+      // Checks if a file pointed by a message relates to the Rust source code and
+      // corrects it if needed and if possible
+      function normalizePath(path) {
+        if (rustSrcPath && path.length > 6) {
+          const prefix = path.substring(0, 7);
+          if (prefix === '../src/' || prefix === '..\\src\\') {
```

Comment:
This function will work in such case because in the line 115 I get rid of `../src` completely and use the value of `RUST_SRC_PATH` instead as the base path.

And `../src` is emitted by rustc independently of one's source location, so I think we could rely on it as the marker.